src/gui/shareusergroupdialog.cpp

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright (C) by Roeland Jago Douma <roeland@owncloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#include "shareusergroupdialog.h"
+#include "ui_shareusergroupdialog.h"
+#include "ui_sharedialogshare.h"
+#include "account.h"
+#include "json.h"
+#include "folderman.h"
+#include "folder.h"
+#include "accountmanager.h"
+#include "theme.h"
+#include "configfile.h"
+#include "capabilities.h"
+
+#include "thumbnailjob.h"
+#include "share.h"
+#include "sharee.h"
+
+#include "QProgressIndicator.h"
+#include <QBuffer>
+#include <QFileIconProvider>
+#include <QClipboard>
+#include <QFileInfo>
+#include <QCompleter>
+
+namespace OCC {
+
+ShareUserGroupDialog::ShareUserGroupDialog(AccountPtr account, const QString &sharePath, const QString &localPath, bool resharingAllowed, QWidget *parent) :
+   QDialog(parent),
+    _ui(new Ui::ShareUserGroupDialog),
+    _account(account),
+    _sharePath(sharePath),
+    _localPath(localPath),
+    _resharingAllowed(resharingAllowed)
+{
+    setAttribute(Qt::WA_DeleteOnClose);
+    setObjectName("SharingDialogUG"); // required as group for saveGeometry call
+
+    _ui->setupUi(this);
+
+    //Is this a file or folder?
+    _isFile = QFileInfo(localPath).isFile();
+
+    _ui->searchPushButton->setEnabled(false);
+    _ui->shareeView->hide();
+    _ui->searchMorePushButton->hide();
+    _ui->sharePushButton->hide();
+
+    _manager = new ShareManager(_account, this);
+    connect(_manager, SIGNAL(sharesFetched(QList<QSharedPointer<Share>>)), SLOT(slotSharesFetched(QList<QSharedPointer<Share>>)));
+    connect(_manager, SIGNAL(shareCreated(QSharedPointer<Share>)), SLOT(getShares()));
+}
+
+void ShareUserGroupDialog::done( int r ) {
+    ConfigFile cfg;
+    cfg.saveGeometry(this);
+    QDialog::done(r);
+}
+
+ShareUserGroupDialog::~ShareUserGroupDialog()
+{
+    delete _ui;
+}
+
+void ShareUserGroupDialog::on_shareeLineEdit_textEdited(const QString &text)
```

Comment:
btw i once had a situation where ctrl+V did not trigger this. something to check out..